mesonbuild/options.py

```diff
@@ -103,8 +102,9 @@ class ArgparseKWs(TypedDict, total=False):
 }
 
 _BAD_VALUE = 'Qwert Zuiopü'
+_optionkey_cache: T.Dict[T.Tuple[str, str, MachineChoice], OptionKey] = {}
```

Comment:
Should this be a class object rather than a module level variable?

Reply:
It cant be because the class is mucking with `__new__`